app/src/main/java/com/marvel/marvel/budget/model/BudgetModelImpl.kt

```diff
@@ -0,0 +1,41 @@
+package com.marvel.marvel.budget.model
+
+
+import com.marvel.marvel.budget.model.resolutionstrategy.Strategy
+import com.marvel.marvel.viewmodel.ComicsViewModel
+import rx.Observable
+import rx.Observer
+import rx.Scheduler
+import rx.Subscription
+
+class BudgetModelImpl(
+    override val comics: List<ComicsViewModel>,
+    private val strategy: Strategy,
+    private val workerThread: Scheduler,
+    private val mainThread: Scheduler
+) : BudgetModel {
+
+    private var subscription: Subscription? = null
+    private var observer: Observer<List<ComicsViewModel>>? = null
+
+    override fun sortComicsListByPrice(budget: String?) {
+        val observable = Observable.fromCallable({
+            strategy.sortComicsByPricePerPage(comics, budget)
+        })
+            .subscribeOn(workerThread)
+            .observeOn(mainThread)
+            .cache()
+        observer?.let { subscription = observable.subscribe(observer) }
```

Comment:
Once you are using a `let`, you should be using an `it` reference inside it. Here it should be:
```
observer?.let { subscription = observable.subscribe(it) }
```
Otherwise, `observer` could be `null` again. Try it with a Kotlin function that doesn't take nullable. Koltin won't be able to make the inference that it's non-null.

Reply:
my bad, thanks for pointing that out